client/lib/user/user.d.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * External Dependencies
+ */
+import { EventEmitter } from 'events';
+import { Language } from '../../languages';
+import { GravatarOptions } from '../../gravatar/types';
+import { URL } from '../../types';
+
+type JSONSerializable = boolean | string | number;
+type WPCOMError = { message: string };
+type WPCOMHeaders = { [ name: string ]: string } & { status: number };
+
+type WPCOMCallback = (
+	error: WPCOMError | null,
+	result: JSONSerializable | null,
+	headers: WPCOMHeaders
+) => void;
+
+export interface User extends EventEmitter {
+	initialize: () => Promise< void >;
+	clearStoreIfChanged: () => void;
+	get: () => UserData;
+	fetch: () => Promise< void >;
+	handleFetchFailure: ( error: Error ) => void;
+	handleFetchSuccess: ( userdata: UserData ) => void;
+	getLanguage: () => Language | void;
```

Comment:
comment lost across updates: possibly `getLanguage: () => Language | undefined` depending on how we feel about indicating the return of the value `undefined`